lavalink/events.py

```diff
@@ -3,60 +3,132 @@ class Event:
 
 
 class QueueEndEvent(Event):
-    """ This event is dispatched when there are no more songs in the queue. """
+    """
+    This event is dispatched when there are no more songs in the queue.
+
+    Parameters
+    ----------
+    player: BasePlayer
+        The player that has no more songs in queue
+    """
     def __init__(self, player):
         self.player = player
 
 
 class TrackStuckEvent(Event):
-    """ This event is dispatched when the currently playing song is stuck. """
-    def __init__(self, player, track, threshold):
+    """
+    This event is dispatched when the currently playing track is stuck.
+
+    Parameters
+    ----------
+    player: BasePlayer
+        The player that has the playing track being stuck
+    track: AudioTrack
+        The track is stuck from playing
+    threshold: int
+        The amount of time the track had while being stuck"""
+    def __init__(self, player, track, threshold: int):
         self.player = player
         self.track = track
         self.threshold = threshold
 
 
 class TrackExceptionEvent(Event):
-    """ This event is dispatched when an exception occurs while playing a track. """
+    """
+    This event is dispatched when an exception occurs while playing a track.
+
+    Parameters
+    ----------
+    player: BasePlayer
+        The player that had the exception occur while playing a track
+    track: AudioTrack
+        The track that had the exception while playing
+    exception: Exception
+        The type of exception that the track had while playing"""
     def __init__(self, player, track, exception):
-        self.exception = exception
         self.player = player
         self.track = track
+        self.exception = exception
 
 
 class TrackEndEvent(Event):
-    """ This event is dispatched when the player finished playing a track. """
-    def __init__(self, player, track, reason):
-        self.reason = reason
+    """
+    This event is dispatched when the player finished playing a track.
+
+    Parameters
+    ----------
+    player: BasePlayer
+        The player that finished playing a track
+    track: AudioTrack
+        The track that finished playing
+    reason: str
+        The reason why the track stopped playing"""
+    def __init__(self, player, track, reason: str):
```

Comment:
i don't think that's necessary, because exceptions are handled internally and don't depend on user input therefore this is not required.